lib/workers/repository/updates/generate.ts

```diff
@@ -181,10 +181,13 @@ export function generateBranchConfig(
       regEx(/to vv(\d)/), // TODO #12071
       'to v$1'
     );
-    if (upgrade.toLowerCase) {
-      // We only need to lowercase the first line
+    if (upgrade.semanticCommitCasing) {
+      // We only need to lowercase the first word
       const splitMessage = upgrade.commitMessage.split('\n');
-      splitMessage[0] = splitMessage[0].toLowerCase();
+      splitMessage[0] = splitMessage[0].replace(
+        splitMessage[0].includes(':') ? regEx(/: \w+/) : regEx(/^\w+/),
```

Comment:
I'm not sure this is accurate enough

Reply:
I think this should be done in commit message builder.

Reply:
```
str = str.replace(
        str.includes(':') ? /^.*: \w+/ : /^\w+/,
        (match) => match.toLowerCase()
      );
```

Reply:
```js
str = str.replace(
        str.includes(':') ? /.*: \w+/ : /^\w+/,
        (match) => match.toLowerCase()
      );
 ```
 This snippet works fine for the all the cases except `Really great feature: Foo`, for this type of message it lowercases the whole string which is a problem. But is such a commit message valid, doesn't it violate the good-commit-message rules.
 
 Regarding the accuracy of the regex, we are only targeting the sub-string we want to lowercase 
 1. /.* :  \w/ -> It will look for a semi-colon and a word after it (i.e first word of message)
 -Eg. `fix: Upgrade something` for this our match is `fix: Upgrade`
 2. /^\w/ -> It matches first word of the message.
 -Eg. `Upgrade Something Please` -> It matches `Upgrade`.
 
 Also I was wondering what would be the updated commit message for `Really great feature: Foo`
 1. `really great feature: Foo` or,
 2. `really great feature: foo`

Reply:
i still suggest to move the whole logic to  this file:

https://github.com/renovatebot/renovate/blob/5af1b77f6919b554b3170c2465094a1442eb0221/lib/workers/repository/model/commit-message.ts#L1-L55

As we build the message there, so it should be more easily to fix it there for all

Reply:
@viceice I am not sure where the changes neerd to be updated in the new file, could you please guide me by commenting on the file where the certain functionalities needs to be added.

Reply:
@RahulGautamSingh I guess in general would be nice if `BranchUpgradeConfig` interface will have instance of `CommitMessage` class. Therefore need to refactor `lib/workers/repository/updates/generate.ts` to using this class. During this refactoring maybe will be required extend functionality of `CommitMessage` or create inherited class for specific purposes.

Reply:
Seems I will have to give more time to go through the code and refactor it. I am closing this PR for now and will create a new one when I have updated the code.
Thanks for the help @pret-a-porter 

Hey @rarkins , I will close this PR and once you give the above suggestion by @pret-a-porter a 👍.